doc/design/parallel-node-config.md

```diff
@@ -0,0 +1,195 @@
+
+## Motivation
+Allow SR-IOV Network Operator to configure more than one node at the moment.
+
+### Use Cases
+
+### Goals
+* Number of drainable at the moment nodes should be 1 be default
+* Number of drainable at the moment nodes should be configured by pool
+* Nodes pool should be defined by node selector
+* A node could be included into several pools. In this case pool should be 
+  selected by it’s priority like it’s implemented for SriovNetworkNodePolicy
+
+
+### Non-Goals
+Parallel NICs configuration on the same node is out of scope of this proposal
+
+## Proposal
+
+Introduce nodes pool drain configuration to meet following requirements:
+* Number of drainable at the moment nodes should be 1 be default
+* Number of drainable at the moment nodes should be configured by pool
+* Nodes pool should be defined by node selector
+* A node could be included into several pools. In this case pool should be selected by it’s priority like it’s
+  implemented for `SriovNetworkNodePolicy`
+
+
+### Workflow Description
+A new Drain controller will be introduced to manage node drain and cordon procedures. That means we don't need to do
+drain and use `drain lock` in config daemon anymore. The overall drain process will be covered by the following states:
+
+```golang
+AnnoIdle            = "Idle"
+AnnoDrainRequired   = "Drain_Required"
+AnnoMcpPaused       = "Draining_MCP_Paused"
+AnnoDraining        = "Draining"
+AnnoDraining        = "Draining_Complete"
+```
+
+Drain controller will watch for node annotation changes, `SriovNetworkPoolConfig` and `SriovNetworkNodeState` changes:
+```golang
+// +kubebuilder:rbac:groups="",resources=nodes,verbs=get;list;watch;update;patch
+// +kubebuilder:rbac:groups="",resources=SriovNetworkPoolConfig,verbs=get;list;watch;update;patch
+// +kubebuilder:rbac:groups="",resources=SriovNetworkNodeState,verbs=get;list;watch;update;patch
+```
+
+Since new `SriovNetworkNodeState` will appear with `syncState=InProgess` controller will `Drain_Required` annotation.
+Another reconcile loop will check if current `Draining` nodes count is less than `maxUnavailable` and will
+mark it as `Draining` to start draining procedure. If current Draining nodes count equals to 
+`maxUnavailable`, no annotation updates will be applied in the current reconcile loop.
+
+If  `SriovNetworkNodeState` will appear with `syncState=Succeed` and node annotation `Draining_Complete` controller
+should change annotation to `Idle`.
+
+We can add a webhook to block the change of the configuration when the nodes are draining to a lower number. Without
+validating webhook controller will proceed drain procedure for the current `Draining` nodes until the count of them
+will decrease to specified `maxUnavailable` number.
+
+We need this logic in controller to have it centralized and not to have raise condition issues during annotation update.
+
+Config daemon will not have drain-related logic with will simplify it a lot.
```

Comment:
it will  have minimal drain related logic ( it still needs to read the annot) and possibly set it when requesting drain.

Reply:
@e0ne here you should say that config daemon will have minimal drain related logic ? because it still needs to set / wait on some annotation.